_posts/2023-12-11-augmented-reality-in-travel-applications.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+title: "Our previous experiments with augmented reality (AR) in travel applications"
+date: 2023-12-11 00:00:00 +0200
+categories: javascript motivation product
+author_name: Adrian Schmid
+read_time: 5
+feature_image: posts/2023-12-11-augmented-reality-in-travel-applications/augmented-reality.jpg
+---
+
+Since its inception, HolidayCheck has thrived on user-generated content (UGC), which remains our cornerstone. Our commitment to elevating this UGC experience continually drives us to innovate for our holidaymakers. In this spirit, we've been exploring fresh avenues, such as integrating interactive, geo-based experiences accessible right from the browser. This exploration opens many possibilities, like offering immersive hotel room previews, comprehensive travel guides, and more to enrich the traveler's journey with us.
+
+## AR with geolocalization in browsers
+
+Fast and pragmatic MVP development on smartphones requires a robust and feature-rich toolkit. [Apple's ARKit](https://developer.apple.com/augmented-reality/arkit/) and [Google's ARCore](https://developers.google.com/ar?hl=en) came into question to support native hardware (i.e. the camera in our case) in the best possible way. Nevertheless, I opted for [AR.js](https://github.com/AR-js-org/AR.js), a browser-based solution, due to its simplicity in terms of quick setup and speed of experimentation. At the time, December 2021, the framework was already almost 2 years young and quite mature.
+
+Even though I was never a fan (and user) of Pokemon Go, many people are familiar with the app that allowed you to use the real world as a game map to catch your Pokemon. But since our vacationers are more interested in sightseeing when they travel, I wanted to develop an interactive camera that shows exciting places nearby.
+
+## The setup
+
+Experiments have a great advantage: you can start a greenfield project. An option that is not always available to developers. In my case, I opted for the following simple setup:
+
+- A simple HTML file containing AR.js markup. We will go into this in more detail in the next section.
+- A JSON: two representative locations to the left and right of my apartment at the time with geo-coordinates coming from our database with over 173,000 POIs: [Augsburg City Hall](https://www.holidaycheck.de/pi/rathaus/24e3daee-7218-3ff1-8111-613f11d0dbc4) and the [Fuggerei](https://www.holidaycheck.de/pi/fuggerei/7a9ada69-b008-31af-8361-0e12f2d62158). What a great way to get to know my home town!
+- A JavaScript function: To load nearby places and enrich them with some data.
+- A simple express server: To deliver the HTML. Why? Because I don't want to run around with my laptop. The app should be accessible from my iPhone.
+
+## Initialize the camera in the browser
+
+For an executable version, you only need markup - no code (except for the AR.js library). The location-based camera needs a scene to which it must be added to work. It looks like this:
```

Comment:
This is purely stylistic and perhaps somewhat semantic.
Instead of using `-` (minus) character, you should use `—` (dash) one.

Rationale:
 - renders differently (especially when you have `-` that is a real minus, and means minus, and should be read as minus)
 - most likely is better interpreted by robots, although I am no expert here
 - it is the character you should use, same as `?` when constructing a question sentence